drizzle-orm/src/pg-core/query-builders/select.ts

```diff
@@ -179,11 +179,12 @@ export abstract class PgSelectQueryBuilder<
 		this.joinsNotNullableMap = typeof this.tableName === 'string' ? { [this.tableName]: true } : {};
 	}
 
-	private createJoin<TJoinType extends JoinType>(
+	private createJoin<TJoinType extends JoinType, TIsLateral extends (TJoinType extends 'left' | 'inner' ? boolean : false)>(
```

Comment:
Why does it only allow using lateral joins for left and inner?

Reply:
I couldn't find documentation on the syntax of lateral joins in Postgres, but by experimenting a bit with the different types of joins combined with `LATERAL`, I got the following Postgres error when I tried a `RIGHT JOIN LATERAL`:
```
The combining JOIN type must be INNER or LEFT for a LATERAL reference.
```

Reply:
I'll try to find more docs about that, but it seems like we can proceed with these two joins for now